includes/RottenLinks.php

```diff
@@ -9,25 +9,40 @@ public static function getResponse( $url ) {
 		$config = $services->getConfigFactory()->makeConfig( 'rottenlinks' );
 
 		// Make the protocol lowercase
-		$urlexp = explode( '://', $url );
+		$urlexp = explode( '://', $url, 2 );
 		$proto = strtolower( $urlexp[0] ) . '://';
 		$site = $urlexp[1];
 		$urlToUse = $proto . $site;
 
+		$status = static::getHttpStatus( $urlToUse, 'HEAD', $services, $config );
+		// Some websites return 4xx or 5xx on HEAD requests but GET with the same URL gives a 200.
+		if ($status >= 400) {
+			$status = static::getHttpStatus( $urlToUse, 'GET', $services, $config );
+		}
+
+		return $status;
+	}
+
+	private static function getHttpStatus( $url, $method, $services, $config ) {
 		$httpProxy = $config->get( 'RottenLinksHTTPProxy' );
 
+		$userAgent = $config->get( 'RottenLinksUserAgent' );
+		if ( is_null( $userAgent ) ) {
+			$userAgent = 'RottenLinks, MediaWiki extension (https://github.com/miraheze/RottenLinks), running on ' . $config->get( 'Server' );
+		}
+
 		$request = $services->getHttpRequestFactory()->createMultiClient( [ 'proxy' => $httpProxy ] )
 			->run( [
-				'url' => $urlToUse,
-				'method' => 'HEAD',
+				'url' => $url,
+				'method' => $method,
 				'headers' => [
-					'user-agent' => 'RottenLinks, MediaWiki extension (https://github.com/miraheze/RottenLinks), running on ' . $config->get( 'Server' )
+					'user-agent' => $userAgent
 				]
 			], [
 				'reqTimeout' => $config->get( 'RottenLinksCurlTimeout' )
 			]
 		);
-
-		return (int)$request['code'];
+		$status = (int)$request['code'];
```

Comment:
Why?

Reply:
You're right, that makes no sense. Let me undo that. I hope I can still add commits to the PR.

Reply:
It's back to the way it was.